x/wasm/module.go

```diff
@@ -199,6 +215,91 @@ func (AppModule) EndBlock(_ sdk.Context, _ abci.RequestEndBlock) []abci.Validato
 	return []abci.ValidatorUpdate{}
 }
 
+func init() {
+	appmodule.Register(
+		&modulev1.Module{},
+		appmodule.Provide(
+			provideModule,
+		),
+	)
+}
+
+type wasmInputs struct {
+	depinject.In
+
+	Config                *modulev1.Module
+	Cdc                   codec.Codec
+	Key                   *store.KVStoreKey
+	AppOpts               servertypes.AppOptions `optional:"true"`
+	WasmOpts              []Option               `optional:"true"`
+	AvailableCapabilities string
+	MsgServiceRouter      *baseapp.MsgServiceRouter
+	GRPCQueryRouter       *baseapp.GRPCQueryRouter
+	AccountKeeper         authkeeper.AccountKeeper
+	BankKeeper            bankkeeper.Keeper
+	StakingKeeper         *stakingkeeper.Keeper
+	DistrKeeper           distrkeeper.Keeper
+
+	IBCKeeper        *ibckeeper.Keeper
+	TransferKeeper   ibctransferkeeper.Keeper
+	ScopedWasmKeeper capabilitykeeper.ScopedKeeper
+
+	// LegacySubspace is used solely for migration of x/params managed parameters
+	LegacySubspace exported.Subspace `optional:"true"`
+}
+
+// Dependency Injection Outputs
+type wasmOutputs struct {
+	depinject.Out
+
+	Module     appmodule.AppModule
+	WasmKeeper *Keeper
+}
+
+func provideModule(in wasmInputs) wasmOutputs {
```

Comment:
this function needs to be public or will cause an error on DI
```suggestion
func ProvideModule(in wasmInputs) wasmOutputs {
```